component-api/src/main/java/org/talend/sdk/component/api/service/cache/LocalCache.java

```diff
@@ -15,51 +15,69 @@
  */
 package org.talend.sdk.component.api.service.cache;
 
+import java.io.Serializable;
+import java.util.function.Predicate;
 import java.util.function.Supplier;
 
 /**
  * Framework service injectable in components or services methods
  * to handle local caching.
  * Useful for actions when deployed in a long running instance
  * when actions are costly.
+ * 
+ * @param <T>  the type of cached data.
  */
-public interface LocalCache {
+public interface LocalCache<T> extends Serializable {
```

Comment:
This cache is not typed so better to not type it since it can end up with a classcastexception by design.